cmd/cli/serve/util.go

```diff
@@ -47,94 +44,131 @@ func GetComputeConfig(
+		TotalResourceLimits:      *totalResources,
+		JobResourceLimits:        *jobResources,
+		DefaultJobResourceLimits: *defaultResources,
+
+		// TODO(review): assumedly we no longer support this feature?
+		// IgnorePhysicalResourceLimits:          false,
+
+		// NB(forrest): not setting these fields should result in a default populating the field by calling method
+		// JobNegotiationTimeout:                 time.Duration(cfg.Node.Compute.JobTimeouts.JobNegotiationTimeout),
+		// MinJobExecutionTimeout:                time.Duration(cfg.Node.Compute.JobTimeouts.MinJobExecutionTimeout),
+		// MaxJobExecutionTimeout:                time.Duration(cfg.Node.Compute.JobTimeouts.MaxJobExecutionTimeout),
+		// DefaultJobExecutionTimeout:            time.Duration(cfg.Node.Compute.JobTimeouts.DefaultJobExecutionTimeout),
+		// JobExecutionTimeoutClientIDBypassList: cfg.Node.Compute.JobTimeouts.JobExecutionTimeoutClientIDBypassList,
+
 		JobSelectionPolicy: node.JobSelectionPolicy{
-			Locality:            semantic.JobSelectionDataLocality(cfg.Node.Compute.JobSelection.Locality),
-			RejectStatelessJobs: cfg.Node.Compute.JobSelection.RejectStatelessJobs,
-			AcceptNetworkedJobs: cfg.Node.Compute.JobSelection.AcceptNetworkedJobs,
-			ProbeHTTP:           cfg.Node.Compute.JobSelection.ProbeHTTP,
-			ProbeExec:           cfg.Node.Compute.JobSelection.ProbeExec,
+			// Locality:            semantic.JobSelectionDataLocality(cfg.Node.Compute.JobSelection.Locality),
+			// TODO(review): assumedly policy should be to accept jobs with data anywhere as we have disabled the option
+			// in the config?
+			Locality:            semantic.Anywhere,
+			RejectStatelessJobs: cfg.JobAdmissionControl.RejectStatelessJobs,
+			AcceptNetworkedJobs: cfg.JobAdmissionControl.AcceptNetworkedJobs,
+			ProbeHTTP:           cfg.JobAdmissionControl.ProbeHTTP,
+			ProbeExec:           cfg.JobAdmissionControl.ProbeExec,
 		},
-		LogRunningExecutionsInterval: time.Duration(cfg.Node.Compute.Logging.LogRunningExecutionsInterval),
-		LogStreamBufferSize:          cfg.Node.Compute.LogStreamConfig.ChannelBufferSize,
-		ExecutionStore:               executionStore,
-		LocalPublisher:               cfg.Node.Compute.LocalPublisher,
-	})
+
+		// NB(forrest): not setting these fields should result in a default populating the field by calling method
+		// LogRunningExecutionsInterval: time.Duration(cfg.Logging.LogDebugInfoInterval),
+		// LogStreamBufferSize:          cfg.Node.Compute.LogStreamConfig.ChannelBufferSize,
+	}
+
+	// TODO(review): assumedly this should work this way, but not sure.
+	// what is the default publisher config in the event one is not provided?
+	if cfg.Publishers.Enabled(types2.KindPublisherLocal) && cfg.Publishers.HasConfig(types2.KindPublisherLocal) {
+		lpcfg, err := types2.DecodeProviderConfig[types2.LocalPublisherConfig](cfg.Publishers)
+		if err != nil {
+			return node.ComputeConfig{}, err
+		}
+		params.LocalPublisher = types.LocalPublisherConfig(lpcfg)
+	}
```

Comment:
@wdbaruni should we abort setting this field if the configuration has disabled the local publisher?

Reply:
yeah, we shouldn't attempt to process or initialize disabled publishers